test/jdk/javax/swing/JFileChooser/HTMLFileName.java

```diff
@@ -0,0 +1,147 @@
+import java.io.File;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 8139228
+ * @summary JFileChooser should not render Directory names in HTML format
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual HTMLFileName
+ */
+
+public class HTMLFileName {
+    static File directory;
+
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                    1. New Directory is created in current directory with name
+                       "<html><h1 color=#ff00ff><font face="Comic Sans MS">Testing Name".
+                    2. On "HTML disabled" frame :
+                          a. Verify that the folder name must be plain text.
+                          b. If the folder name in file pane window and also in directory
+                             ComboBox remains in plain text, then test PASS.
+                             If it appears to be in HTML format with Pink color, then test FAILS
+                             (Verify for all Look and Feel).
+                    3. On "HTML enabled" frame :
+                          a. Verify that the folder name remains in HTML
+                             format with name "Testing Name" pink in color.
+                          b. If the folder name in file pane window and also in directory
+                             ComboBox remains in HTML format string, then test PASS.
+                             If it appears to be in plain text, then test FAILS.
+                             (Verify for all Look and Feel).
+                """;
+
+        try {
+            PassFailJFrame.builder()
+                    .title("Test Instructions")
+                    .instructions(INSTRUCTIONS)
+                    .columns(45)
+                    .testUI(initialize())
+                    .position(PassFailJFrame.Position.TOP_LEFT_CORNER)
+                    .build()
+                    .awaitAndCheck();
+        } finally {
+            if (directory != null) {
+                directory.delete();
+            }
+        }
+    }
+
+    public static List<JFrame> initialize() {
+        boolean isWindows = System.getProperty("os.name").startsWith("Windows");
+        File homeDir = FileSystemView.getFileSystemView().getHomeDirectory();
+        String fileName = homeDir + File.separator +
+                "<html><h1 color=#ff00ff><font face=\"Comic Sans MS\">Testing Name";
+        directory = new File(fileName);
+        directory.mkdir();
+        JFileChooser jfc;
+        JFileChooser jfc_HTML_Enabled;
+        JFrame frame_HTML_Enabled = new JFrame("HTML enabled");
+        JFrame frame = new JFrame("HTML disabled");
+        if (isWindows) {
+            jfc = new JFileChooser(new VirtualFileSystemView());
+            jfc_HTML_Enabled = new JFileChooser(new VirtualFileSystemView());
+        } else {
+            jfc = new JFileChooser(homeDir);
+            jfc_HTML_Enabled = new JFileChooser(homeDir);
+        }
```

Comment:
I suggest creating a method that creates one file chooser placed in a frame; pass a boolean parameter to specify whether to set the `"html.disable"` property to `false` or keep its default value of `true`.

Then your `initialize` method would be as simple as a single return statement:

```java
    private static List<JFrame> initialize() {
        return List.of(createFileChooser(true), createFileChooser(false));
    }
```

Reply:
Updated.